stablehlo/reference/Element.cpp

```diff
@@ -638,6 +638,19 @@ Element log(const Element &el) {
       [](std::complex<double> e) { return std::log(e); });
 }
 
+Element log_plus_one(const Element &el) {
+  return mapWithUpcastToDouble(
+      el, [](double e) { return std::log1p(e); },
+      [](std::complex<double> e) {
+        // log1p(a+bi) = log((a+1)+bi)
+        //             = 0.5 * log((a+1)^2 +  b^2) + i atan2(b, (a+1))
+        return std::complex<double>(
```

Comment:
Simpler:
```
return mapWithUpcastToDouble(
      el, [](double e) { return std::log1p(e); },
      [](std::complex<double> e) {
        return std::log(e + std::complex<double>(1.0));
      });
```

Reply:
Hi @karthikrangasai 
Can you please point to where you get this formula from?

```
// log1p(a+bi) = log((a+1)+bi)
        //             = 0.5 * log((a+1)^2 +  b^2) + i atan2(b, (a+1))
```

Reply:
Hello @sdasgup3 , I derived it as follows:

$\log1p(a + b \iota)$

= $\log((a+1) + b \iota)$

= $\log(r * (cos\theta + \iota sin\theta))$ where $\mathbf{r} = \sqrt((a+1)^2 + b^2)$ and $\theta = \arctan(\frac{b}{a+1})$

= $\log(r) + log(cos\theta + \iota sin\theta)$

= $\log(\sqrt((a+1)^2 + b^2)) + log( \exp^{\iota \theta})$

= $\frac{1}{2} * log((a+1)^2 + b^2) + \iota \theta$

= $\frac{1}{2} * log((a+1)^2 + b^2) + \iota \arctan(\frac{b}{a+1})$

Reply:
Thanks @karthikrangasai for the explanation.
As suggested by @ghpvnist, let us keep the simplest implementation with

 $\log1p(a + b \iota)$ = $\log((a + b \iota) + (1 + 0 \iota))$

For both https://numpy.org/doc/stable/reference/generated/numpy.log.html and https://numpy.org/doc/stable/reference/generated/numpy.log1p.html
```
For complex-valued input, log (or logp1) is a complex analytical function that has a branch cut [-inf, 0] and is continuous from above on it.
```

I think the approximating the logp1 using log will lose some precision owing to additional rounding of `x + 1`, but the branch cut  for the multiple-valued function (logp1) is properly taken care of.

Reply:
Sure @sdasgup3 . Thanks for explaining why we are using the simpler implementation because I did not know that there could be precision issues in the output values.